logger/src/error.rs

```diff
@@ -0,0 +1,159 @@
+//! Ergonomic log-and-ignore syntax.
+
+/// Trait used to pipeline the common log-and-ignore pattern by
+/// providing better handling of the process, and a clearer more
+/// expressive syntax.
+pub trait Logged {
+    /// Log the provided value. Useful for logging objects in a chain
+    /// of events, or before the `?` operator.
+    #[must_use]
+    fn logged(self) -> Self;
+
+    /// Ignore the provided error.
+    ///
+    /// # Motivation
+    ///
+    /// Sometimes errors cannot be sensibly handled, they can only be
+    /// reported to the user. If the error isn't handled further this
+    /// allows one to cheaply explain to the user of the debug build
+    /// **why** a particular error cannot be ignored.  More
+    /// importantly when one sees an explicit
+    /// ```rust
+    /// err.logged().ignore("I can't handle this error because of X")
+    /// ```
+    /// the reader of the code doesn't need to be provided
+    /// with a comment as to why the error message was ignored
+    #[inline]
+    fn ignored(self, reason: &str)
```

Comment:
What about passing log level inside? Instead of selecting one when deriving.

Reply:
I suppose I could add 
```rust
fn logged_as(self, level: tracing::Level) -> Self {
	event!(level, "{self}");
}

fn ignored_at(self, level: …) {
	…
```
but it doesn't seem to be useful in many cases. It makes the call much more cumbersome. Moreover, there's a `promote` method.